### PR TITLE
Formatting the source code to comply with the LLVM standard

### PR DESCRIPTION
**Pull request by nicholaslegrand102**
I created a .clang-format file that complies with the LLVM standard that was used to format all of the code. 

Closes issue #28

### CODE REVIEW

**Comment by felix642** [comment]
@nicholaslegrand102 you've requested my review on this PR, but this is still marked as a draft. I'm a bit confused, am I supposed to review this PR or not? 🙃

File: src/.clang-format [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+Language:        Cpp
```
Comment (felix642) [inline src/.clang-format:2]: Hmm, that's not right

File: src/.clang-format [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# BasedOnStyle:  LLVM
+AccessModifierOffset: -2
```
Comment (felix642) [inline src/.clang-format:4]: This is not something that exists in C.

File: src/.clang-format [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInConditionalStatement: false
+SpacesInContainerLiterals: false
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+SpaceBeforeSquareBrackets: false
+Standard:        Latest
+UseCRLF:         false
```
Comment (felix642) [inline src/.clang-format:56]: I do not think we need this option.

File: src/.clang-format [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+BreakStringLiterals: true
+ColumnLimit:     80
+ContinuationIndentWidth: 4
+IncludeBlocks:   Merge
+IndentCaseLabels: false
+IndentWidth:     2
+IndentWrappedFunctionNames: false
+KeepEmptyLinesAtTheStartOfBlocks: true
+MaxEmptyLinesToKeep: 1
+PointerAlignment: Right
```
Comment (felix642) [inline src/.clang-format:42]: I vote to align pointers in the middle. @bisson2000 is agreeing with me.

**Review by felix642 — CHANGES_REQUESTED** [review]
Hi @nicholaslegrand102 I left a few comments. Would you mind taking a look ? :)

File: src/.clang-format (hunk 2 above)
Comment (nicholaslegrand102) [reply src/.clang-format:4]: good point

File: src/.clang-format (hunk 3 above)
Comment (nicholaslegrand102) [reply src/.clang-format:56]: Why not?

**Review by felix642 — APPROVED** [review]
LGTM 👍🏿

**Review by bisson2000 — APPROVED** [review]
lgtm

**Comment by felix642** [comment]
Thank you @nicholaslegrand102. I have merged your PR in the main branch. Feel free to open a PR on the official repo now.

**Pull request merged**